File: umm/src/main/java/com/wyy/service/UaaClient.java
package com.wyy.service;

import com.wyy.client.AuthorizedFeignClient;
import com.wyy.dto.Message;
import com.wyy.dto.MessageDraft;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@AuthorizedFeignClient(name = "uaa")
public interface UaaClient {

    @RequestMapping(value ="/api/messages/send", method = RequestMethod.POST)
    ResponseEntity<Message> sendMessage(@RequestBody Message message);

    @RequestMapping(value ="/api/messages/multicast", method = RequestMethod.POST)
    ResponseEntity<Void> sendMulticastMessage(@RequestBody MessageDraft messageDraft);

}

File: umd-python/app/service/uaa_client.py
from app.service.http_client import http_client


service_name = 'uaa'


def send_message(message, jwt):
    body = {
        'action': 'send_message',
        'args': {
            'message': message,
        }
    }
    return http_client(service_name, body=body, jwt=jwt)
